controllers/productController.js

```diff
@@ -68,4 +68,27 @@ const createProduct = (req, res) => {
     .catch((err) => console.log(err))
 }
 
-module.exports = { getProducts, getProductById, getProductsByGroup, getProductsByUserId, createProduct }
+const putAnswer = (req, res) => {
+  console.log('req.body', req.body)
+  Product.updateOne(
+    { _id: req.body.docId, 'fixedQandAs.question': fixedQuestions[req.body.questionIdx] },
+    {
+      $push: {
+        'fixedQandAs.$.answers': {
+          userId: req.user._id,
+          description: req.body.answer,
+          report: {
+            wrong: 0,
+            affiliate: 0,
+            threats: 0,
+            privacy: 0,
+          },
+        },
+      },
+    }
+  )
```

Comment:
updateOne is not recomended in mongoose. Use findOne() and save() instead.
[https://mongoosejs.com/docs/documents.html#updating-using-save](https://mongoosejs.com/docs/documents.html#updating-using-save)